src/utils.ts

```diff
@@ -166,3 +166,27 @@ export async function genEnvFile(buildDir: string) {
     // Ignore errors, Meson could be too old to support --dump-format.
   }
 }
+
+export async function patchCompileCommands(buildDir: string) {
+  // Remove ccache from compile commands because they confuse Intellisense:
+  // https://github.com/microsoft/vscode-cpptools/issues/7616
+  try {
+    var filePath = path.join(buildDir, "compile_commands.json")
+    var db = JSON.parse(fs.readFileSync(filePath, "utf-8"));
+  } catch {
+    return;
+  }
```

Comment:
I don't think `path.join` is going to throw here, it just returns a string like python's `os.path.join()`.

`JSON.parse()` will throw a syntax error if the json is invalid, which seems unlikely, but it's possible.

I think we might be better off using the asynchronous read function though, since compile_commands.json can be quite large.

So I think that would look something like:
```typescript
const filePath = path.join(buildDir, "compile_commands.json");
let db = await fs.readFile(filePath, "utf-8", (err, data): Error | object => {
  if (err) { return err; }
  try {
    return JSON.parse(data);  /// might need to add `as object` here...
  } catch (e) {
    return e;
  }
});

if (db instanceof Error) { return; }
```

Reply:
Even better, we already had parseJSONFileIfExists() for that.